update proto and format

fardream opened the pull request.

fardream committed:
update proto and format

fardream committed:
add bazel support

fardream committed:
remove one layer of mod

fardream committed:
make proto public

yuhanfang commented on WORKSPACE:
```diff
@@ -0,0 +1 @@
+workspace(name = "com_github_foonetic_crosslink_protocol")
```
I don't think bazel belongs in this repo since nothing is built with it. We can revisit this if we need to build cross language tooling. Can you use https://bazel.build/rules/lib/repo/git#new_git_repository in the downstream bazel project instead?

yuhanfang commented on crosslink.proto:
```diff
@@ -166,21 +166,22 @@ message LookupInstrumentIdRequest {
+  message LookupLocationIdError {
+    string error = 1;
+  }
+  LookupLocationIdError error = 2;
```
I agree with wrapping the error behind a more general type. But will we ever return both a location and an error? If not it seems that wrapping with oneof makes the API clearer.

yuhanfang commented on crosslink.proto:
```diff
@@ -80,7 +80,7 @@ message TargetPosition {
   DecimalValue quantity = 4;
   // the instrument to trade for.
-  int32 base_instrument = 5;
+  int64 base_instrument = 5;
```
Should this just be unsigned?

yuhanfang commented on crosslink.proto:
```diff
@@ -112,7 +112,7 @@ message Fill {
   // router then routes the order to SERUM, and gets the fills. The destination
   // should be SERUM, where the fill is ultimately from.
-  int32 fill_venue = 5;
+  int64 fill_venue = 5;
```
Should this just be unsigned?

yuhanfang commented on crosslink.proto:
```diff
@@ -166,21 +166,22 @@ message LookupInstrumentIdRequest {
-  oneof response {
-      int32 instrument_id = 1;
-      string error = 2;
+  int64 instrument_id = 1;
```
Should this just be unsigned?

fardream committed:
update to service:

- remove `target_position_id` and `signature`.
- add `SendTargetPosition` and `GetCurrentposition`

fardream committed:
rework target position workflow

- sending target position and receiving updates are separated.
- dedicated cancel method.

fardream committed:
add readme

leina05 commented on crosslink.proto:
```diff
@@ -128,59 +79,135 @@ message Cancel {}
-  rpc TargetPosition(stream TargetPositionRequest)
-      returns (stream TargetPositionResponse) {}
+  // Subscribe Target Position Updates
+  rpc SubscribeTargetPosition(SubscribeTargetPositionRequest)
```
One question: should the `SubscribeTargetPositionResponse` stream ever close? Or will it stay open until the client drops the connection? If it stays open, I would consider renaming to `SubscribePosition`, since it's tied to the position generally rather than a specific target.

fardream committed:
rename to `SubscribePosition` and allow disconnect

fardream committed:
futher clean up proto, add timestamps to fill and position.
